sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -234,6 +234,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     }
     def mapChild(child: Any): Any = child match {
       case arg: TreeNode[_] if containsChild(arg) => mapTreeNode(arg)
+      // CaseWhen Case or any tuple type
+      case tuple: (_, _) => (mapChild(tuple._1), mapChild(tuple._2))
```

Comment:
Let's make better use of pattern matching here, and deconstruct the tuple there, e.g.: `case (left, right) => (mapChild(left), mapChild(right))`

Reply:
This just fixes tuple, we should audit all expressions and make sure there are no other weird patterns.

Reply:
okay I will test all other expressions, and raise separate PR (or umbrella jira) if required

Reply:
I double checked , only the `tuple` and `Option` were missing.